tests/responses/test_fileresponse.py

```diff
@@ -25,6 +25,60 @@ def test_content_length_buffer(self):
         response = FileResponse(io.BytesIO(b'binary content'))
         self.assertEqual(response.headers['Content-Length'], '14')
 
+    def test_content_length_nonzero_starting_position_file(self):
+        file = open(__file__, 'rb')
+        file.seek(10)
+        response = FileResponse(file)
+        self.assertEqual(response.headers['Content-Length'], str(os.path.getsize(__file__) - 10))
+        response.close()
+
+    def test_content_length_nonzero_starting_position_buffer(self):
+        test_tuples = (
+            ('BytesIO', io.BytesIO),
+            ('UnseekableBytesIO', UnseekableBytesIO),
+        )
+        for buffer_class_name, BufferClass in test_tuples:
+            with self.subTest(buffer_class_name=buffer_class_name):
+                buffer = BufferClass(b'binary content')
+                buffer.seek(10)
+                response = FileResponse(buffer)
+                self.assertEqual(response.headers['Content-Length'], '4')
+
+    def test_content_length_nonzero_starting_position_file_seekable_no_tell(self):
+        class TestFile:
+            def __init__(self, path, *args, **kwargs):
+                self._file = open(path, *args, **kwargs)
+
+            def read(self, n_bytes=-1):
+                return self._file.read(n_bytes)
+
+            def seek(self, offset, whence=io.SEEK_SET):
+                return self._file.seek(offset, whence)
+
+            def seekable(self):
+                return True
+
+            @property
+            def name(self):
+                return self._file.name
+
+            def close(self):
+                if self._file:
+                    self._file.close()
+                    self._file = None
+
+            def __enter__(self):
+                return self
+
+            def __exit__(self, e_type, e_val, e_tb):
+                self.close()
+
+        file = TestFile(__file__, 'rb')
```

Comment:
that test is checking the support for a possible object that is seekable but has no `.tell()`.
`tempfile.NamedTemporaryFile()` has `.tell()`.

Reply:
Thanks. Still wondering though if you can pass it by name to your TestFile wrapper. You'd be operating on a temp file but your TestFile instance would lack 'seek', no?

Reply:
Ah, I see you're adhering to the way these tests were already using '__file__' -- no need to increase the scope of this PR more. Apologies.

Reply:
> Thanks. Still wondering though if you can pass it by name to your TestFile wrapper. You'd be operating on a temp file but your TestFile instance would lack 'tell', no?

ohh, that's what you meant.
well, that would be opening an already opened file, and when i try it gives me a 'permission denied' error, so no.
however, i can just directly save the `tmp` as TestFile's `._file` and it seems to work fine.
I just need some content for the temporary file.
the thing is, while i agree working on `__file__` may not be the best idea, i don't think having to create/fill a temporary one with each test is a way to go. could we just create a temp file for the duration of the entire test module with `setUpClass()` and `tearDownClass()`? i've never done it before, but it seems to work.

> Ah, I see you're adhering to the way these tests were already using '**file**' -- no need to increase the scope of this PR more. Apologies.

frankly, i'm not sure i know what you mean here, i think you misunderstood me after all.
i wasn't referring in any place to the fact that those tests mostly used `__file__` already (and that i just expanded on them).
but i guess i can agree that we probably should play around with changing everything to working on temp file in another PR.
...if that is what you mean now :p

Reply:
> this test is checking the support for a possible object that is seekable but has no `.tell()`.
> `tempfile.NamedTemporaryFile()` has `.tell()`.

Just as an aside, whether there is any change here or not, but you could do:

```python
with tempfile.NamedTemporaryFile() as tmp:
    del tmp.tell  # Emulate seekable file handle without .tell().
    ...
```

Reply:
Nice, I like Nick's idea.

My answer was going to be, yes, let's defer the temp file idea to another PR, because my original goal was to help you get this marked ready for checkin--however, if you want to run with Nick's idea, that could potentially reduce the overhead of having to create the `TestFile` class, and a smaller diff will get reviewed/merged faster. Up to you.

Reply:
I like the idea, but...
```python
with tempfile.NamedTemporaryFile() as f:
    print(hasattr(f, 'tell'), f.tell)
    del f.tell
    print(hasattr(f, 'tell'), f.tell)
    del f.tell
    del f.tell
    
True <function BufferedRandom.tell at 0x0000000004F38EE0>
True <function BufferedRandom.tell at 0x0000000004F38EE0>
Traceback (most recent call last):
  File "<input>", line 6, in <module>
AttributeError: tell
```
Huston, we have a problem. Seems that trying to access that method recreates it.

Also when trying to delete `.tell` of regular file returned by open(), it throws such exception immediately.